### ISSUE TITLE
Closing this, removing and reinstalling fixed the issue :S

### ISSUE THREAD

**Issue by MikeiLL** [weprovide/valet-plus#366]
Closing this, removing and reinstalling fixed the issue :S
`composer global remove weprovide/valet-plus && composer global require weprovide/valet-plus`

_Originally posted by @emateu in https://github.com/weprovide/valet-plus/issues/263#issuecomment-436413926_

This seems to have reared it's head again. I'm running fourth round of remove/require on High Sierra:

```
Command "fix" is not defined.  
                                 

mikekilmer@jamaaladeen-8:~$ composer global remove weprovide/valet-plus && composer global require weprovide/valet-plus
Changed current directory to /Users/mikekilmer/.composer
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 0 installs, 0 updates, 15 removals
  - Removing weprovide/valet-plus (v2.2.1)
  - Removing tightenco/collect (v5.4.33)
  - Removing symfony/process (v3.4.28)
  - Removing symfony/polyfill-mbstring (v1.11.0)
  - Removing symfony/debug (v4.3.0)
  - Removing symfony/console (v3.4.28)
  - Removing psr/simple-cache (1.0.1)
  - Removing psr/log (1.1.0)
  - Removing psr/container (1.0.0)
  - Removing php-di/invoker (1.3.3)
  - Removing nategood/httpful (0.2.20)
  - Removing mnapoli/silly (1.5.1)
  - Removing illuminate/contracts (v5.7.28)
  - Removing illuminate/container (v5.7.8)
  - Removing container-interop/container-interop (1.2.0)
Writing lock file
Generating autoload files
Changed current directory to /Users/mikekilmer/.composer
Using version ^2.2 for weprovide/valet-plus
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 21 installs, 0 updates, 0 removals
  - Installing symfony/polyfill-php72 (v1.11.0): Downloading (100%)         
  - Installing symfony/polyfill-mbstring (v1.11.0): Downloading (100%)         
  - Installing symfony/var-dumper (v4.3.1): Downloading (100%)         
  - Installing tightenco/collect (v5.8.21): Downloading (100%)         
  - Installing nategood/httpful (0.2.20): Downloading (100%)         
  - Installing symfony/process (v4.3.1): Downloading (100%)         
  - Installing psr/container (1.0.0): Downloading (100%)         
  - Installing php-di/invoker (2.0.0): Downloading (100%)         
  - Installing symfony/service-contracts (v1.1.2): Downloading (100%)         
  - Installing symfony/polyfill-php73 (v1.11.0): Downloading (100%)         
  - Installing symfony/console (v4.3.1): Downloading (100%)         
  - Installing mnapoli/silly (1.7.1): Downloading (100%)         
  - Installing symfony/translation-contracts (v1.1.2): Downloading (100%)         
  - Installing symfony/translation (v4.3.1): Downloading (100%)         
  - Installing nesbot/carbon (2.19.1): Downloading (100%)         
  - Installing psr/simple-cache (1.0.1): Downloading (100%)         
  - Installing illuminate/contracts (v5.8.20): Downloading (100%)         
  - Installing doctrine/inflector (v1.3.0): Downloading (100%)         
  - Installing illuminate/support (v5.8.20): Downloading (100%)         
  - Installing illuminate/container (v5.8.20): Downloading (100%)         
  - Installing weprovide/valet-plus (v2.2.1): Downloading (100%)         
symfony/var-dumper suggests installing ext-intl (To show region name in time zone dump)
symfony/service-contracts suggests installing symfony/service-implementation
symfony/console suggests installing symfony/event-dispatcher
symfony/console suggests installing symfony/lock
symfony/console suggests installing psr/log (For using the console logger)
symfony/translation suggests installing symfony/config
symfony/translation suggests installing symfony/yaml
illuminate/support suggests installing illuminate/filesystem (Required to use the composer class (5.8.*).)
illuminate/support suggests installing moontoast/math (Required to use ordered UUIDs (^1.1).)
illuminate/support suggests installing ramsey/uuid (Required to use Str::uuid() (^3.7).)
illuminate/support suggests installing vlucas/phpdotenv (Required to use the env helper (^3.3).)
Writing lock file
Generating autoload files
mikekilmer@jamaaladeen-8:~$ valet fix
Password:

                                 
  Command "fix" is not defined.  
                                 

mikekilmer@jamaaladeen-8:~$ valet
Laravel Valet 2.2.1

Usage:
  command [options] [arguments]

Options:
  -h, --help            Display this help message
  -q, --quiet           Do not output any message
  -V, --version         Display this application version
      --ansi            Force ANSI output
      --no-ansi         Disable ANSI output
  -n, --no-interaction  Do not ask any interactive question
  -v|vv|vvv, --verbose  Increase the verbosity of messages: 1 for normal output, 2 for more verbose output and 3 for debug

Available commands:
  fetch-share-url    Get the URL to the current Ngrok tunnel
  forget             Remove the current working (or specified) directory from Valet's list of paths
  help               Displays help for a command
  install            Install the Valet services
  link               Link the current working directory to Valet
  links              Display all of the registered Valet links
  list               Lists commands
  on-latest-version  Determine if this is the latest version of Valet
  open               Open the site for the current (or specified) directory in your browser
  park               Register the current working (or specified) directory with Valet
  paths              Get all of the paths registered with Valet
  restart            Restart the Valet services
  secure             Secure the given domain with a trusted TLS certificate
  share              Generate a publicly accessible URL for your project
  start              Start the Valet services
  stop               Stop the Valet services
  tld                [domain] Get or set the TLD used for Valet sites.
  trust              Add sudoers files for Brew and Valet to make Valet commands run without passwords
  uninstall          Uninstall the Valet services
  unlink             Remove the specified Valet link
  unsecure           Stop serving the given domain over HTTPS and remove the trusted TLS certificate
  use                Change the version of php used by valet
  which              Determine which Valet driver serves the current working directory
mikekilmer@jamaaladeen-8:~$ composer global remove weprovide/valet-plus && composer clearcache && composer global require weprovide/valet-plus
Changed current directory to /Users/mikekilmer/.composer
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 0 installs, 0 updates, 21 removals
  - Removing weprovide/valet-plus (v2.2.1)
  - Removing tightenco/collect (v5.8.21)
  - Removing symfony/var-dumper (v4.3.1)
  - Removing symfony/translation-contracts (v1.1.2)
  - Removing symfony/translation (v4.3.1)
  - Removing symfony/service-contracts (v1.1.2)
  - Removing symfony/process (v4.3.1)
  - Removing symfony/polyfill-php73 (v1.11.0)
  - Removing symfony/polyfill-php72 (v1.11.0)
  - Removing symfony/polyfill-mbstring (v1.11.0)
  - Removing symfony/console (v4.3.1)
  - Removing psr/simple-cache (1.0.1)
  - Removing psr/container (1.0.0)
  - Removing php-di/invoker (2.0.0)
  - Removing nesbot/carbon (2.19.1)
  - Removing nategood/httpful (0.2.20)
  - Removing mnapoli/silly (1.7.1)
  - Removing illuminate/support (v5.8.20)
  - Removing illuminate/contracts (v5.8.20)
  - Removing illuminate/container (v5.8.20)
  - Removing doctrine/inflector (v1.3.0)
Writing lock file
Generating autoload files
Cache directory does not exist (cache-vcs-dir): 
Clearing cache (cache-repo-dir): /Users/mikekilmer/.composer/cache/repo
Clearing cache (cache-files-dir): /Users/mikekilmer/.composer/cache/files
Clearing cache (cache-dir): /Users/mikekilmer/.composer/cache
All caches cleared.
Changed current directory to /Users/mikekilmer/.composer
Using version ^2.2 for weprovide/valet-plus
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 21 installs, 0 updates, 0 removals
  - Installing symfony/polyfill-php72 (v1.11.0): Downloading (100%)         
  - Installing symfony/polyfill-mbstring (v1.11.0): Downloading (100%)         
  - Installing symfony/var-dumper (v4.3.1): Downloading (100%)         
  - Installing tightenco/collect (v5.8.21): Downloading (100%)         
  - Installing nategood/httpful (0.2.20): Downloading (100%)         
  - Installing symfony/process (v4.3.1): Downloading (100%)         
  - Installing psr/container (1.0.0): Downloading (100%)         
  - Installing php-di/invoker (2.0.0): Downloading (100%)         
  - Installing symfony/service-contracts (v1.1.2): Downloading (100%)         
  - Installing symfony/polyfill-php73 (v1.11.0): Downloading (100%)         
  - Installing symfony/console (v4.3.1): Downloading (100%)         
  - Installing mnapoli/silly (1.7.1): Downloading (100%)         
  - Installing symfony/translation-contracts (v1.1.2): Downloading (100%)         
  - Installing symfony/translation (v4.3.1): Downloading (100%)         
  - Installing nesbot/carbon (2.19.1): Downloading (100%)         
  - Installing psr/simple-cache (1.0.1): Downloading (100%)         
  - Installing illuminate/contracts (v5.8.20): Downloading (100%)         
  - Installing doctrine/inflector (v1.3.0): Downloading (100%)         
  - Installing illuminate/support (v5.8.20): Downloading (100%)         
  - Installing illuminate/container (v5.8.20): Downloading (100%)         
  - Installing weprovide/valet-plus (v2.2.1): Downloading (100%)         
symfony/var-dumper suggests installing ext-intl (To show region name in time zone dump)
symfony/service-contracts suggests installing symfony/service-implementation
symfony/console suggests installing symfony/event-dispatcher
symfony/console suggests installing symfony/lock
symfony/console suggests installing psr/log (For using the console logger)
symfony/translation suggests installing symfony/config
symfony/translation suggests installing symfony/yaml
illuminate/support suggests installing illuminate/filesystem (Required to use the composer class (5.8.*).)
illuminate/support suggests installing moontoast/math (Required to use ordered UUIDs (^1.1).)
illuminate/support suggests installing ramsey/uuid (Required to use Str::uuid() (^3.7).)
illuminate/support suggests installing vlucas/phpdotenv (Required to use the env helper (^3.3).)
Writing lock file
Generating autoload files
mikekilmer@jamaaladeen-8:~$ which valet
/usr/local/bin/valet
mikekilmer@jamaaladeen-8:~$ ls -l /usr/local/bin/valet
lrwxr-xr-x  1 mikekilmer  wheel    61B Jun  6 16:52 /usr/local/bin/valet@ -> /Users/mikekilmer/.composer/vendor/weprovide/valet-plus/valet
mikekilmer@jamaaladeen-8:~$ valet fix
Password:

                                 
  Command "fix" is not defined.  
                                 
```

**Comment by MikeiLL** [comment]
I have also spent some time in [this issue](https://github.com/weprovide/valet-plus/issues/131) yesterday and at one point seemed to have it working, but MySql connections weren't.

**Comment by cmubo** [comment]
I'm having the same problem.

my `./composer/vendor/weprovide/valet-plus/cli/valet ` doesn't replicate the one in this repo and has no fix commands inside it. It's also only 304 lines long.

but the file still ends on.

`$app->run();`

So I don't think its an unfinished/corrupt download. 

When doing a comparison between the cli directory inside this repo and the cli directory that composer is installing on my computer, there are a lot of differences.

**Comment by MikeiLL** [comment]
I seem to have 

> I'm having the same problem.
> 
> my `./composer/vendor/weprovide/valet-plus/cli/valet ` doesn't replicate the one in this repo and has no fix commands inside it. It's also only 304 lines long.
> 
> but the file still ends on.
> 
> `$app->run();`
> 
> So I don't think its an unfinished/corrupt download.
> 
> When doing a comparison between the cli directory inside this repo and the cli directory that composer is installing on my computer, there are a lot of differences.

My `~/.composer/vendor/weprovide/valet-plus/cli/valet.php` file is also 304 lines long.

I was wondering (before I went to sleep) what one would do to install the framework directly with Git as opposed to Composer. I guess you would clone it into `~/.composer/vendor/weprovide` but that might break other functionality.

The https://github.com/laravel/valet/blob/master/cli/valet.php file is 374 lines long (v2.3.3).

**Comment by cmubo** [comment]
I know I shouldn't have done this but I did it anyway and got the fix command back, just breaking my PHP installation when using valet install in the process (due to apcu issues #131 didn't seem to fix it but that's my problem).  I wouldn't recommend doing this, I'm just letting you know that I tried it. 

So I replaced everything in the `./composer/vendor/weprovide/valet-plus/` directory with everything in this current repository, it will complain (or at least for me) about the `SebastianBergmann\Version` in 
`/cli/valet.php `

commented out line 16
`//use SebastianBergmann\Version;`

replaced line 25 & 27 with the composer version
```
$version = new Version('1.0.27' , __DIR__ . '/../');

$app = new Application('Valet+', $version->getVersion());
```

with 

```
$version = '2.2.1';

$app = new Application('Laravel Valet', $version);
```


As I say, I wouldn't recommend this, I'm just letting you know my findings. 

TL;DR
Got valet fix back but it breaks valet install.

**Comment by MikeiLL** [comment]
> $version = '2.2.1';
> 
> $app = new Application('Laravel Valet', $version);

Which fix did you do from the #131 thread? Also If you wouldn't mind explaining what is happening with the replacements on lines 25 and 27 in the `cli/valet.php` script, I'll appreciate the insight.

**Comment by cmubo** [comment]
@MikeiLL 

> Please remove the top 3 lines and save the file. Remove all extension= lines from the top of the file. Then show me `php -v`.

just make sure you are looking in the right php.ini file, instead of the php directory, look in the valet-php directory.

My Path  `usr/local/etc/valet-php/7.1/php.ini`

![Ive7ViRA](https://user-images.githubusercontent.com/7483407/59150464-063eb100-8a1c-11e9-8d6c-8ccdaacff976.png)

I've commented them out in the screenshot. 

> Also If you wouldn't mind explaining what is happening with the replacements on lines 25 and 27 in the cli/valet.php script, I'll appreciate the insight.

This was literally just replacing the instances of the version function which was pulled in by 
`use SebastianBergmann\Version;`

Definitely not meant to be a long time fix, but it does look to me like what composer is downloading is an older version of laravel valet, I think the current version is 2.3.4.

**Comment by MikeiLL** [comment]
Thank you so much. How did you find your way to the valet-php directory. When I type which php I get 

```
$ which php
/usr/local/bin/php
mikekilmer@jamaaladeen-8:~$ ls -l /usr/local/bin/php
lrwxr-xr-x  1 mikekilmer  wheel    38B Jun  8 13:26 /usr/local/bin/php@ -> ../Cellar/valet-php@7.1/7.1.26/bin/php
```

Is it just because you know that we installed the various versions of php with homebrew and that the `php.ini` files for `../Cellar/valet-php@7.1/7.1.26/bin/php` are stored at `../Cellar/valet-php@7.1/{some version}/bin/php`? Or is there a config somewhere in `/usr/local/Cellar/valet-php\@7.1/7.1.26/` that points to it?

**Comment by cmubo** [comment]
Good Question, it was just by chance that I had been messing around with the `usr/local/etc/` directory enough and noticed the valet-php directory, remembering that the installation seemed to have its own PHP version that it installed.

I'm a front-end dev so this is all new to me, I'd like to know if there is a specific place to get the information too though if anyone could answer. Hopefully, this thread will be picked up soon.

**Comment by MikeiLL** [comment]
https://gist.github.com/vitorbritto/0555879fe4414d18569d#gistcomment-2930338 the links in this post might help. I'm on round 5 or so of reinstalling mysql5.7.

From `RecursiveDirectoryIterator::__construct(/usr/local/var/mysql): failed to open dir: No such file or directory` to https://github.com/weprovide/valet-plus/issues/262#issuecomment-499295511 to `Setting password for root user failed. ` again.

[As of OS X Yosemite, /etc/hostconfig is no longer installed as part of the OS.](https://stackoverflow.com/a/33276979/2223106). Should there be one there?

**Comment by samgranger** [comment]
Looks like a colleague pushed the wrong tags so Laravel's Valet was being installed. See #369
